cmd/restic/lock.go

```diff
@@ -43,7 +43,38 @@ func lockRepository(ctx context.Context, repo restic.Repository, exclusive bool)
 		lockFn = restic.NewExclusiveLock
 	}
 
-	lock, err := lockFn(ctx, repo)
+	var lock *restic.Lock
+	var err error
+
+	retrySleep := 5 * time.Second
```

Comment:
Retrying every five seconds, appears to be a bit frequent, especially if a longer RetryLock duration is used. What about increasing the duration depending on the RetryLock duration?

Reply:
I have to admit I have no educated guess for those values.

What about we start with 5s and bump the sleep duration at every loop until we reach the max of 5 minutes ?
Some sort of exponential back off ?

Is 5m too long ? Should we reduce the max value to 1 minute ?

Reply:
A maximum of 1 minute is probably better. Waiting for 5 minutes requires quite a bit of patience. 

What might be useful is to bound the value of `retrySleepNext` such that it does not exceed the retryLock limit (but only if the duration doesn't be come too short).

Reply:
> What might be useful is to bound the value of retrySleepNext such that it does not exceed the retryLock limit (but only if the duration doesn't be come too short).

I am not sure to understand this one. From what I understand we don't need to reduce `retrySleepNext` below `retryLock` because the `retryTimeout` (based on retryLock) will exit before the sleeping period.

We do make sure `retrySleep` is smaller than `retryLock` before entering the loop.